public/app/features/alerting/unified/state/actions.ts

```diff
@@ -0,0 +1,26 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { RulesSourceResult } from 'app/types/unified-alerting/internal';
+import { fetchRules } from '../api/rules';
+import { getAllDataSources } from '../utils/config';
+import { DataSourceType } from '../utils/datasource';
+
+/*
+ * Will need to be updated to:
+ *
+ * 1. Fetch grafana managed rules when the endpoint becomes available
+ * 2. Reconcile with rules from the ruler where ruler is available
+ */
+
+export const fetchRulesAction = createAsyncThunk(
```

Comment:
so this was poorly thought out by me. If request for one of the rules sources is slow/hangs, user has to wait for it to finish before being able to view any of the rules. 
Need to load rules for each rules source separately, back to the drawing board

Reply:
done